code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -42,36 +45,73 @@
 			break
 
 		if(isliving(A))
-			var/mob/living/L = A
-			if((L.faction == src.faction) && !attack_same)
-				continue
-			else if(L in friends)
-				continue
-			else
-				if(!L.stat)
-					stance = HOSTILE_STANCE_ATTACK
-					T = L
-					break
+			T = find_weakest()

```

Comment:
This is a weird point to call this. Surely you'd replace the entire loop starting at line 37 with this function call...? Consider: this loop (the one starting at line 37) is made to iterate through the targets and locate a valid one. And `find_weakest` is made to do the exact same thing.

So this entire loop should be replaced with `T = find_weakest()`.